python-checks/src/main/java/org/sonar/python/checks/cdk/DisabledESDomainEncryptionCheck.java

```diff
@@ -0,0 +1,82 @@
+
+@Rule(key = "S6308")
+public class DisabledESDomainEncryptionCheck extends AbstractCdkResourceCheck {
+  private static final String OMITTING_MESSAGE = "Omitting %s causes encryption of data at rest to be disabled for this {OpenSearch|Elasticsearch} domain." +
+    " Make sure it is safe here.";
+  private static final String UNENCRYPTED_MESSAGE = "Make sure that using unencrypted {OpenSearch|Elasticsearch} domains is safe here.";
+  private static final String ENABLED = "enabled";
+
+  @Override
+  protected void registerFqnConsumer() {
+    checkFqn("aws_cdk.aws_opensearchservice.Domain", DisabledESDomainEncryptionCheck::checkNormalDomain);
+    checkFqn("aws_cdk.aws_opensearchservice.CfnDomain", DisabledESDomainEncryptionCheck::checkCfnDomain);
+  }
+
+  private static void checkNormalDomain(SubscriptionContext ctx, CallExpression callExpression) {
+    checkDomain(ctx, callExpression, "encryption_at_rest", "aws_cdk.aws_opensearchservice.EncryptionAtRestOptions");
+  }
+
+  private static void checkCfnDomain(SubscriptionContext ctx, CallExpression callExpression) {
+    checkDomain(ctx, callExpression, "encryption_at_rest_options", "aws_cdk.aws_opensearchservice.CfnDomain.EncryptionAtRestOptionsProperty");
+  }
+
+  private static void checkDomain(SubscriptionContext ctx, CallExpression callExpression, String encryptionArgName, String argFqnMethod) {
+    getArgument(ctx, callExpression, encryptionArgName).ifPresentOrElse(
+      argEncryptedTrace ->
+        argEncryptedTrace.addIssueIf(expr -> isMethodCallWithoutEnabledOrSetToFalse(ctx, expr, argFqnMethod)
+          || (expr.is(Tree.Kind.DICTIONARY_LITERAL) && isDictionaryEnabledValueFalseOrAbsent(ctx, (DictionaryLiteral) expr))
+          , UNENCRYPTED_MESSAGE)
+      , () -> ctx.addIssue(callExpression.callee(), String.format(OMITTING_MESSAGE, encryptionArgName))
+    );
+  }
+
+  private static boolean isMethodCallWithoutEnabledOrSetToFalse(SubscriptionContext ctx, Expression expr, String argFqnMethod) {
+    if (!AbstractCdkResourceCheck.isFqnValue(expr, argFqnMethod)) {
+      return false;
+    }
+    if (!expr.is(Tree.Kind.CALL_EXPR)) {
+      return true;
+    }
+
+    return getArgument(ctx, (CallExpression) expr, ENABLED)
+      .filter(argEnabledTrace -> !argEnabledTrace.hasExpression(AbstractCdkResourceCheck::isFalse)).isEmpty();
+  }
+
+  private static boolean isDictionaryEnabledValueFalseOrAbsent(SubscriptionContext ctx, DictionaryLiteral dictionaryLiteral) {
+    return dictionaryLiteral.elements().stream()
+      .filter(element -> element.is(Tree.Kind.KEY_VALUE_PAIR)).map(KeyValuePair.class::cast)
+      .filter(keyValuePair -> keyValuePair.key().is(Tree.Kind.STRING_LITERAL)
+        && ((StringLiteral) keyValuePair.key()).trimmedQuotesValue().equals(ENABLED))
+      .map(keyValuePair -> ArgumentTrace.build(ctx, keyValuePair.value()))
+      .allMatch(argTrace -> argTrace.hasExpression(AbstractCdkResourceCheck::isFalse));
+  }
```

Comment:
Such long streams over maps are quite hard to read. Sometimes it's easier to create a small and simple subclass like `Option` which already trims keys, and resolves values when constructed. So you can focus in the stream on filtering the content, not on the preparation of it.